More robust error logging

SChinchi opened the pull request:
- Log a message if the command requires being in a run. This mostly applies while being in a lobby pregame. An edge case I haven't managed to address is being on the report screen, e.g., after dying, as it still counts being in a run, but a lot of the commands will fail with NPE or ArgumentOutOfRangeIndexException.
- Don't silently use the default value for an optional argument when parsing fails, but log a parsing error.
- Be explicit when a command has not been given enough arguments, e.g., `"Insufficient args" + Lang.RELEVANT_COMMAND_ARGS`.

Also gathered all the command help texts in Lang.cs for convenience.

xiaoxiao921 commented on Code/DT-Commands/Lists.cs:
```diff
@@ -233,21 +232,21 @@ private static void CCListTeams(ConCommandArgs args)
                 sb.AppendLine($"{card.spawnCard.name}");
             }
-            Log.Message(sb);
+            Log.MessageNetworked(sb.ToString(), args, LogLevel.MessageClientOnly);
```
Why message networked there?

xiaoxiao921 replied:
Ok on a second thought I wonder if it's not better to always try to network it properly by merging Message and MessageNetworked

xiaoxiao921 replied:
Yeah list commands are always local (I'm simply judging from the ConVarFlags) so that's why I found it odd, that said con commands have a sender.isLocalPlayer so we could use it to determine if the log need to be networked or not automatically

SChinchi replied:
I will revert these changes for now and if we decide to decide to change the behaviour of Message and MessageNetworked, it can be done in a future PR. Just a heads up, there was originally a message networked in loadout_set_skin_variant, even though it has ConVarFlags.None.

xiaoxiao921 commented:
Yeah I can't verify right now but maybe that's an additional check we could do

SChinchi commented:
It does turn out to be the body being null while dead.

This has far-reaching consequences for some commands, such as, `god`.  If we just have a check to skip dead players, then in multiplayer it can cause god mode to be out of sync for everyone. So either we apply some hook to flip the god mode on dead players when they respawn, or we make sure the command only has an effect when everyone on the team is alive. This is a behavioral change and this PR was only intended to check for and log errors. So I would rather leave this bug in and sort it out in another PR.

But for commands like `give_item`, where accessing the inventory of a dead body will lead to an NPE, it would be in the spirit of this PR to just log a "body dead" message and end prematurely.

xiaoxiao921 commented:
Depending on where the nre happens, this can be gracefully handled by just checking if args.sender is dead or not, no?

SChinchi commented:
I also had a look at MessageNetworked and it seems to check for the sender being a local player before calling Message internally, so I didn't revert those modifications. Could you please verify whether my understanding is correct?

I fixed the dead body bug on most commands. However, I didn't touch `god`, `midgame`, and `lategame`, which will require some modifications if the effect is supposed to apply to everyone at the same time or not at all. If this is something we aren't going to address in this PR and there aren't any other outstanding issues, consider this complete from my side.

SChinchi commented:
I don't know what autocomplete voodoo I unleashed with the second to last commit, but I seemed to have changed all `args[i].ToUpper() == Lang.PINGED` to not convert to uppercase, so I reverted that. In the meantime, I had a look at the source code and I realised using the CharacterMaster instead of CharacterBody addresses any previous issues as the former is not null on character death. I added that in with the current commit because it was a minimal change.
